tests/v2/validation/encryption/rke2_secret_encryption_test.go

```diff
@@ -0,0 +1,168 @@
+	subSession := s.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := s.client.WithSession(subSession)
+	require.NoError(s.T(), err)
+
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(s.T(), clusterName, "Cluster name is not provided")
+
+	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
+	require.NoError(s.T(), err)
+
+	steveClient, err := client.Steve.ProxyDownstream(clusterID)
+	require.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), steveClient)
+
+	secretTemplate := secrets.NewSecretTemplate(namegen.AppendRandomString("test-secret"), s.ns, map[string][]byte{secretKey: []byte(secretValue)}, v1.SecretTypeOpaque)
+	secretObj, err := steveClient.SteveType(secrets.SecretSteveType).Create(secretTemplate)
+	require.NoError(s.T(), err)
+	logrus.Infof("secret created successfully!")
+
+	envFrom := []v1.EnvFromSource{
+		{
+			SecretRef: &v1.SecretEnvSource{
+				LocalObjectReference: v1.LocalObjectReference{Name: secretObj.Name},
+			},
+		},
+	}
+	wload := namegen.AppendRandomString("test-workload")
+	containerTemplate := workloads.NewContainer("ranchertest", "ranchertest/mytestcontainer", v1.PullAlways, []v1.VolumeMount{}, envFrom)
+	podTemplate := workloads.NewPodTemplate([]v1.Container{containerTemplate}, []v1.Volume{}, []v1.LocalObjectReference{}, nil)
+	deploymentTemplate := workloads.NewDeploymentTemplate(wload, s.ns, podTemplate, true, nil)
+
+	_, err = steveClient.SteveType(workloads.DeploymentSteveType).Create(deploymentTemplate)
+	require.NoError(s.T(), err)
+
+	podResults, podErrors := pods.StatusPods(client, clusterID)
+	assert.NotEmpty(s.T(), podResults)
+	assert.Empty(s.T(), podErrors)
+
+	kubeConfig, err := kubeconfig.GetKubeconfig(client, clusterID)
+	require.NoError(s.T(), err)
+
+	restConfig, err := (*kubeConfig).ClientConfig()
+	require.NoError(s.T(), err)
+
+	pods, err := steveClient.SteveType(pods.PodResourceSteveType).List(nil)
+	require.NoError(s.T(), err)
+
+	cmd := []string{
+		"/bin/sh",
+		"-c",
+		"echo $testKey",
+	}
+
+	for _, pod := range pods.Data {
+		if strings.HasPrefix(pod.ObjectMeta.Name, wload) {
+			err := kwait.Poll(5*time.Second, 2*time.Minute, func() (done bool, err error) {
+				output, err := kubeconfig.KubectlExec(restConfig, pod.ObjectMeta.Name, s.ns, cmd)
+				if err != nil {
+					return false, nil
+				}
+				require.Equal(s.T(), output.String(), secretValue)
+				logrus.Info("environment variable "+secretKey+" value : ", output.String())
+				return true, nil
+			})
+			require.NoError(s.T(), err)
+			break
+		}
+	}
+
+	nodesList, err := steveClient.SteveType("node").List(nil)
+	require.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), nodesList)
+
+	_, err = client.ReLogin()
+	require.NoError(s.T(), err)
+
+	for _, node := range nodesList.Data {
+		result, err := strconv.ParseBool(node.Labels[nodes1.EtcdNodeLabel])
+		if result && node.Annotations[nodes1.ClusterAnnotation] == clusterName && err == nil {
```

Comment:
why is error check in this if statement? I think it should be a separate assert just above this one for clarity and correct failure if this is an error.